{{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.plugin_name.lower().replace(' ', '') }}plugin.py

```diff
@@ -10,56 +10,99 @@
 {%- set widget_name = cookiecutter.plugin_name.replace(' ', '') + 'Widget' -%}
 {%- set config_page = cookiecutter.plugin_name.replace(' ', '') + 'ConfigPage' %}
 
+{%- if cookiecutter.graphical_plugin == 'n' %}
 from qtpy.QtWidgets import QVBoxLayout
-
-{% if cookiecutter.spyder3_compatibility == 'y' -%}
+{%- endif %}
+{% if cookiecutter.spyder3_compatibility == 'y' %}
 try:
     from spyder.api.plugins import SpyderPluginWidget
 except ImportError:
     from spyder.plugins import SpyderPluginWidget # Spyder3
     {%- if cookiecutter.create_config_page == 'y' %}
 try:
-    from spyder.api.preferences import
+    from spyder.api.preferences import PluginConfigPage
 except ImportError:
     from spyder.plugins.configdialog import PluginConfigPage # Spyder3
     {%- endif %}
-{% else -%}
+{% else %}
+    {%- if cookiecutter.graphical_plugin == 'y' %}
 from spyder.api.plugins import SpyderPluginWidget
+    {%- else %}
+from spyder.api.plugins import SpyderPlugin
+    {%- endif %}
     {%- if cookiecutter.create_config_page == 'y' %}
 from spyder.api.preferences import PluginConfigPage
-    {%- endif %}
-{%- endif -%}
-
+    {% endif %}
+{%- endif %}
 from .widgets.{{ cookiecutter.plugin_name.lower().replace(' ', '') }}gui import {{ widget_name }}
-{% if cookiecutter.create_config_page == 'y' %}
 
+{% if cookiecutter.create_config_page == 'y' %}
 class {{ config_page }}(PluginConfigPage):
     """{{ cookiecutter.plugin_name }} plugin preferences."""
     pass
 {% endif %}
+{%- if cookiecutter.graphical_plugin == 'n' %}
+class {{ cookiecutter.plugin_name.replace(' ', '') }}Plugin(SpyderPlugin):
```

Comment:
This needs to inherit from QObject first, then from SpyderPlugin. See for instance:

https://github.com/spyder-ide/spyder/blob/master/spyder/api/completion.py